app/app.rb

```diff
@@ -0,0 +1,107 @@
+require 'sinatra'
+require 'sinatra/base'
+require 'sinatra/flash'
+require './app/data_mapper_setup'
+
+class Chitter < Sinatra::Base
+
+  enable :sessions
+  register Sinatra::Flash
+
+  get '/' do
+    if current_user
+      redirect '/sessions/new'
+    else
+      erb :'users/sign_up'
+    end
+  end
+
+  get '/:username' do
+    @user = User.first(username: params[:username])
+    if @user
+      @peeps = @user.peeps.all(:order => :time_stamp.desc )
+      all_users_peep_ids = @peeps.map{ |peep| peep.id }
```

Comment:
Space missing to the left of {.<br>Pass `&:id` as an argument to `map` instead of a block.